azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -788,28 +922,34 @@ String getFlowContainerDiskRequest(final Map<String, String> flowParam) {
    * requested resource):
    * 1) if resource 1 >= resource 2, return 1;
    * 2) if resource 1 < resource 2, return -1;
-   * 3) if resource 1 cannot be compared with resource 2, e.g. parse error, mis-matching resource
-   * types, return 0;
+   * 3) if resource 1 cannot be compared with resource 2, e.g. parse error, return 0;
    * The format of a Kubernetes quantity indicates the type of resource, e.g. cpu, memory.
    * @param resource1
    * @param resource2
    * @return
    */
-  private int compareResources (final String resource1, final String resource2) {
+  private static int compareResources(final String resource1, final String resource2) {
     try {
       final Quantity quantity1 = new Quantity(resource1), quantity2 = new Quantity(resource2);
-      if (quantity1.getFormat() == quantity2.getFormat()) {
-        return quantity1.getNumber().compareTo(quantity2.getNumber()) < 0 ? -1 : 1;
-      }
-    } catch (final QuantityFormatException qe) { // only user requested resource in flow
+      return quantity1.getNumber().compareTo(quantity2.getNumber()) < 0 ? -1 : 1;
+    } catch (final Exception e) { // only user requested resource in flow
       // parameter could result in exception
-      logger.error("QuantityFormatException while parsing user requested resource: " + resource2);
+      logger.error("Exception while parsing requested resource: " + resource1 + "," + resource2);
     }
     // Resources cannot be compared, e.g. resources are different (cpu vs. memory) or
     // exception encountered when parsing resource string
     return 0;
   }
 
+  private static boolean isValidResource(final String resource) {
+    try {
+      new Quantity(resource);
+    } catch (final Exception e) {
```

Comment:
QuantityFormatException should be enough.

Reply:
There will be other types of exceptions thrown from here even though they are not documented by the Quantity() api doc. e.g.
ArithmeticException